pkg/queue/certificate/watcher_test.go

```diff
@@ -70,7 +70,7 @@ func TestCertificateRotation(t *testing.T) {
 
 	// CertWatcher should return the new certificate
 	// Give CertWatcher some time to update the certificate
-	if err := wait.Poll(1*time.Second, 30*time.Second, func() (bool, error) {
+	if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
```

Comment:
According to https://github.com/knative/serving/blob/b6d4f6182871ccd17fc720f12a5bc5ff34705aad/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go#L434-L435
 `wait.Poll()` will not retry when the func returns an error. So I think we should change the following line to

```go
		if san != updatedSAN {
		        t.Logf("CertWatcher did not return the expected certificate. want: %s, got: %s", updatedSAN, san)
			return false, nil
		}
```

? Otherwise, `wait.Poll` will end as soon as `san != updatedSAN` happens?

Reply:
Aww, thanks. Seems like I had the wrong assumption about this method. I tested it now with a very short interval, now it works as expected.